src/solvers/refinement/string_refinement.cpp

```diff
@@ -896,6 +874,44 @@ exprt string_refinementt::substitute_array_with_expr(
   }
 }
 
+/// Fill an array represented by a list of with_expr by propagating values to
+/// the left. For instance `ARRAY_OF(12) WITH[2:=24] WITH[4:=42]` will give
+/// `{ 24, 24, 24, 42, 42 }`
```

Comment:
This should indeed be an ID_array expression and `with_exprt` is expected at the input so I think the documentation correctly reflects the intention of the function